app/controllers/users_controller.rb

```diff
@@ -8,22 +8,32 @@ def show
 
   # GET /users/1/edit
   def edit
+    @tab_select = params[:tab]
+    if @tab_select.present?
+      respond_to do |format|
+        format.js { render action: 'field_select' }
+      end
+    else
+      render :edit
+    end
   end
 
   # PATCH/PUT /users/1
   def update
-    if @user.update(user_params)
-      redirect_to @user, notice: 'User was successfully updated.'
-    else
-      flash[:error] = "An error prohibited your Profile from being saved: #{@user.errors.full_messages.join('. ')}."
-      render :edit
+    respond_to do |format|
+      if @user.update(user_params)
+        flash[:notice] = 'User was successfully updated.'
+      else
+        flash[:error] = "An error prohibited your Profile from being saved: #{@user.errors.full_messages.join('. ')}."
+      end
+      format.js
```

Comment:
Why is the format.js necessary at all as you didn't change the form to use AJAX?